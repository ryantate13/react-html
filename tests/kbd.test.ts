import {kbd} from '../src';

describe('kbd', () => {
    it('renders with children', () => {
        expect(kbd({}, 'TEST')).toMatchSnapshot();
    });
    
    it('renders arrays of children', () => {
        expect(kbd({}, Array(10).fill('TEST'))).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(kbd({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(kbd()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(kbd({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(kbd({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(kbd({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(kbd({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(kbd({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(kbd({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(kbd({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(kbd({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(kbd({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(kbd({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(kbd({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(kbd({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(kbd({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(kbd({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(kbd({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden=true', () => {
        expect(kbd({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden=false', () => {
        expect(kbd({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(kbd({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(kbd({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(kbd({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(kbd({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(kbd({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(kbd({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(kbd({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(kbd({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(kbd({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(kbd({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(kbd({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(kbd({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(kbd({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope=true', () => {
        expect(kbd({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope=false', () => {
        expect(kbd({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(kbd({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(kbd({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(kbd({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(kbd({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(kbd({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(kbd({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style={"color":"#000000"}', () => {
        expect(kbd({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex=1', () => {
        expect(kbd({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(kbd({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(kbd({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(kbd({translate: "no"})).toMatchSnapshot();
    });
});