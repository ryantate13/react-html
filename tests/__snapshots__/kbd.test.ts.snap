// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kbd renders arrays of children 1`] = `
<kbd>
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
</kbd>
`;

exports[`kbd renders with accessKey="a,b,c,d" 1`] = `
<kbd
  accessKey="a,b,c,d"
/>
`;

exports[`kbd renders with autoCapitalize="characters" 1`] = `
<kbd
  autoCapitalize="characters"
/>
`;

exports[`kbd renders with autoCapitalize="none" 1`] = `
<kbd
  autoCapitalize="none"
/>
`;

exports[`kbd renders with autoCapitalize="off" 1`] = `
<kbd
  autoCapitalize="off"
/>
`;

exports[`kbd renders with autoCapitalize="on" 1`] = `
<kbd
  autoCapitalize="on"
/>
`;

exports[`kbd renders with autoCapitalize="sentences" 1`] = `
<kbd
  autoCapitalize="sentences"
/>
`;

exports[`kbd renders with autoCapitalize="words" 1`] = `
<kbd
  autoCapitalize="words"
/>
`;

exports[`kbd renders with children 1`] = `
<kbd>
  TEST
</kbd>
`;

exports[`kbd renders with className="foo bar baz" 1`] = `
<kbd
  className="foo bar baz"
/>
`;

exports[`kbd renders with contentEditable="false" 1`] = `
<kbd
  contentEditable="false"
/>
`;

exports[`kbd renders with contentEditable="true" 1`] = `
<kbd
  contentEditable="true"
/>
`;

exports[`kbd renders with dir="auto" 1`] = `
<kbd
  dir="auto"
/>
`;

exports[`kbd renders with dir="ltr" 1`] = `
<kbd
  dir="ltr"
/>
`;

exports[`kbd renders with dir="rtl" 1`] = `
<kbd
  dir="rtl"
/>
`;

exports[`kbd renders with draggable="false" 1`] = `
<kbd
  draggable="false"
/>
`;

exports[`kbd renders with draggable="true" 1`] = `
<kbd
  draggable="true"
/>
`;

exports[`kbd renders with empty attributes 1`] = `<kbd />`;

exports[`kbd renders with hidden=false 1`] = `
<kbd
  hidden={false}
/>
`;

exports[`kbd renders with hidden=true 1`] = `
<kbd
  hidden={true}
/>
`;

exports[`kbd renders with id="TEST" 1`] = `
<kbd
  id="TEST"
/>
`;

exports[`kbd renders with inputMode="decimal" 1`] = `
<kbd
  inputMode="decimal"
/>
`;

exports[`kbd renders with inputMode="email" 1`] = `
<kbd
  inputMode="email"
/>
`;

exports[`kbd renders with inputMode="none" 1`] = `
<kbd
  inputMode="none"
/>
`;

exports[`kbd renders with inputMode="numeric" 1`] = `
<kbd
  inputMode="numeric"
/>
`;

exports[`kbd renders with inputMode="search" 1`] = `
<kbd
  inputMode="search"
/>
`;

exports[`kbd renders with inputMode="tel" 1`] = `
<kbd
  inputMode="tel"
/>
`;

exports[`kbd renders with inputMode="text" 1`] = `
<kbd
  inputMode="text"
/>
`;

exports[`kbd renders with inputMode="url" 1`] = `
<kbd
  inputMode="url"
/>
`;

exports[`kbd renders with is="foo" 1`] = `
<kbd
  is="foo"
/>
`;

exports[`kbd renders with itemID="https://google.com/" 1`] = `
<kbd
  itemID="https://google.com/"
/>
`;

exports[`kbd renders with itemProp="https://google.com/" 1`] = `
<kbd
  itemProp="https://google.com/"
/>
`;

exports[`kbd renders with itemRef="foo bar" 1`] = `
<kbd
  itemRef="foo bar"
/>
`;

exports[`kbd renders with itemScope=false 1`] = `
<kbd
  itemScope={false}
/>
`;

exports[`kbd renders with itemScope=true 1`] = `
<kbd
  itemScope={true}
/>
`;

exports[`kbd renders with itemType="https://google.com/" 1`] = `
<kbd
  itemType="https://google.com/"
/>
`;

exports[`kbd renders with lang="" 1`] = `
<kbd
  lang=""
/>
`;

exports[`kbd renders with lang="es-419" 1`] = `
<kbd
  lang="es-419"
/>
`;

exports[`kbd renders with slot="TEST" 1`] = `
<kbd
  slot="TEST"
/>
`;

exports[`kbd renders with spellCheck="false" 1`] = `
<kbd
  spellCheck="false"
/>
`;

exports[`kbd renders with spellCheck="true" 1`] = `
<kbd
  spellCheck="true"
/>
`;

exports[`kbd renders with style={"color":"#000000"} 1`] = `
<kbd
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`kbd renders with tabIndex=1 1`] = `
<kbd
  tabIndex={1}
/>
`;

exports[`kbd renders with title="TEST" 1`] = `
<kbd
  title="TEST"
/>
`;

exports[`kbd renders with translate="no" 1`] = `
<kbd
  translate="no"
/>
`;

exports[`kbd renders with translate="yes" 1`] = `
<kbd
  translate="yes"
/>
`;

exports[`kbd renders without attributes 1`] = `<kbd />`;
